include/zephyr/mgmt/mcumgr/callbacks.h

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef H_MCUMGR_CALLBACKS_
+#define H_MCUMGR_CALLBACKS_
+
+#include <inttypes.h>
+#include <zephyr/sys/slist.h>
+#include <mgmt/mgmt.h>
+
+#ifdef CONFIG_MCUMGR_CMD_FS_MGMT
+#include <zephyr/mgmt/mcumgr/fs_mgmt_callbacks.h>
+#endif
+
+#ifdef CONFIG_MCUMGR_CMD_IMG_MGMT
+#include <zephyr/mgmt/mcumgr/img_mgmt_callbacks.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief MCUmgr callback API
+ * @defgroup mcumgr_callback_api MCUmgr callback API
+ * @ingroup mcumgr
+ * @{
+ */
+
+/** @cond INTERNAL_HIDDEN */
+/** Event which signfies that all event IDs for a particular group should be enabled. */
+#define MGMT_EVT_OP_ID_ALL 0xffff
+
+/** Get event for a particular group and event ID. */
+#define MGMT_DEF_EVT_OP_ID(group, event_id) ((group << 16) | BIT(event_id))
+
+/** Get event used for enabling all event IDs of a particular group. */
+#define MGMT_DEF_EVT_OP_ALL(group) ((group << 16) | MGMT_EVT_OP_ID_ALL)
+/** @endcond  */
+
+/** Get group from event. */
+#define MGMT_EVT_GET_GROUP(event) ((event >> 16) & MGMT_EVT_OP_ID_ALL)
+
+/** Get event ID from event. */
+#define MGMT_EVT_GET_ID(event) (event & MGMT_EVT_OP_ID_ALL)
+
+/**
+ * @typedef mgmt_cb
+ * @brief Function to be called on MGMT notification/event.
+ *
+ * This callback function is used to notify an application or system about a mcumgr mgmt event.
+ *
+ * @param event		MGMT_EVT_OP_[...].
+ * @param failed	Will be set to true if a previous handler has already returned a failure
```

Comment:
It is registration order dependent, yes. Updated.

Reply:
And regarding my question: "What is an example scenario where passing status code between handlers is useful?", do we anticipate any specific scenario or we're adding more flexibility just in case?

Reply:
For me, no, I don't see the usage for it. For example in the file access callback, if there are 5 registered handlers and the 3rd one denies it, I don't see the point in handler 4 and 5 being called. As for if others see use in it I do not know, Dominik's idea is that handlers that deal with getting notifications only might want it e.g. a logger